components/freertos/tasks.c

```diff
@@ -4097,6 +4108,16 @@ static portTASK_FUNCTION( prvIdleTask, pvParameters )
 #endif /* configUSE_TICKLESS_IDLE */
 /*-----------------------------------------------------------*/
 
+
+#if CONFIG_COMPILER_STACK_MIRROR
+    IRAM_ATTR __attribute__((no_instrument_function))
+    esp_stack_mirror_t * pvTaskGetStackMirrorPointer()
+    {
+        TCB_t *pxTCB = xTaskGetCurrentTaskHandle();
```

Comment:
Thats an interesting approach, and not one I would have thought of! I agree your approach seems better. The perf of my approach is *just* good enough. It would be ideal to improve perf.